internal/services/network/network_interface_resource.go

```diff
@@ -325,6 +355,22 @@ func resourceNetworkInterfaceUpdate(d *pluginsdk.ResourceData, meta interface{})
 		},
 	}
 
+	if d.HasChange("auxiliary_mode") {
+		if auxiliaryMode, hasAuxiliaryMode := d.GetOk("auxiliary_mode"); hasAuxiliaryMode {
+			update.Properties.AuxiliaryMode = pointer.To(networkinterfaces.NetworkInterfaceAuxiliaryMode(auxiliaryMode.(string)))
+		}
+	} else {
+		update.Properties.AuxiliaryMode = existing.Model.Properties.AuxiliaryMode
+	}
+
+	if d.HasChange("auxiliary_sku") {
+		if auxiliarySku, hasAuxiliarySku := d.GetOk("auxiliary_sku"); hasAuxiliarySku {
+			update.Properties.AuxiliarySku = pointer.To(networkinterfaces.NetworkInterfaceAuxiliarySku(auxiliarySku.(string)))
+		}
+	} else {
+		update.Properties.AuxiliarySku = existing.Model.Properties.AuxiliarySku
+	}
```

Comment:
how do users unset this property?

Reply:
If users remove this property from the .tf, `hasAuxiliarySku = false` so that `update.Properties.AuxiliarySku` will not be assigned a value, then Azure will unset this property.

Reply:
However the Go SDK for this defines this field as `omitempty` (due to it being optional in the API)

> AuxiliarySku                *NetworkInterfaceAuxiliarySku       `json:"auxiliarySku,omitempty"`

Meaning that if it's `nil` no value will be sent to the API - meaning this can't be cleared out. So can we add a test for this? This may need to be explicitly set to an empty string to handle this use-case

Reply:
The testcase has already existed in this PR and it can pass. In the testcase, the workflow is [unset]-> [set] -> [unset].

![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/76987228/61a58ad8-4110-49b0-bf68-e1f08ccbd448)